modules/redhen_connection/src/Plugin/Action/ChangeConnectionRole.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Drupal\redhen_connection\Plugin\Action;
+
+/**
+ * Change the role of a Redhen Connection.
+ *
+ * @Action(
+ *   id = "connection_change_role_action",
+ *   label = @Translation("Change the role for the selected connection(s)"),
+ *   type = "redhen_connection"
+ * )
+ */
+class ChangeConnectionRole extends ChangeConnectionRoleBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute($connection = NULL) {
+    $role = $this->configuration['role'];
+    $mismatch = FALSE;
+    // Skip changing the role for the connection if already present,
+    // OR if the role is not available for the given connection.
+    $entity_storage = \Drupal::service('entity_type.manager')->getStorage('redhen_connection_role');
+    $roles = array_keys($entity_storage->loadByProperties(['connection_type' => $connection->getType()]));
+    if ($connection !== FALSE && $connection->get('role', $role)->getString() != $role) {
```

Comment:
What's up with the second argument for `->get()` here? I assume this is `ContentEntityBase::get()`, which only takes 1 argument, so it's probably just throwing the `$role` one away?
```suggestion
    if ($connection !== FALSE && $connection->get('role')->getString() != $role) {
```

Reply:
good catch, that was a misplaced variable. Yes, I imagine it was just throwing that extra argument away because I wasn't seeing any exceptions or errors.